app/classifier/tasks/survey/editor.cjsx

```diff
@@ -190,35 +278,72 @@ module.exports = React.createClass
       </label>
     </div>
 
-  handleFiles: (forEachRow, afterFileHook, e) ->
+  handleImportTabs: (tab) ->
+    @setState editing: tab
+
+  handleApply: ->
+    if @state.choicesName or @state.confusionsName
+      @props.task.choices = @state.task.choices
+      @props.task.choicesOrder = @state.task.choicesOrder
+      @props.task.questionsMap = @state.task.questionsMap
+    if @state.characteristicsName
+      @props.task.characteristics = @state.task.characteristics
+      @props.task.characteristicsOrder = @state.task.characteristicsOrder
+      @props.task.choices = @state.task.choices
+    if @state.questionsName
+      @props.task.questions = @state.task.questions
+      @props.task.questionsOrder = @state.task.questionsOrder
+      @props.task.questionsMap = @state.task.questionsMap
+    @props.workflow.update('tasks').save()
+      .then =>
+        @clearState()
+
```

Comment:
these if-tests don't necessarily make sense to me; what does confusionsName have to do with questionsMap, etc?